lib/src/formatters/shared/baseExport.ts

```diff
@@ -0,0 +1,181 @@
+    content: string | Record<string, unknown>
+  ): void;
+
+  protected async fetchAPIData() {
+    await this.fetchVariants();
+    const textItemsMap = await this.fetchTextItemsMap();
+    const componentsMap = await this.fetchComponentsMap();
+
+    return { textItemsMap, componentsMap };
+  }
+
+  /**
+   * For each project/variant permutation and its fetched .strings data,
+   * create a new file with the expected naming
+   *
+   * @returns {OutputFile[]} List of Output Files
+   */
+  protected transformAPIData(data: ExportFormatAPIData): TOutputFile[] {
+    Object.entries(data.textItemsMap).forEach(
+      ([projectId, projectVariants]) => {
+        Object.entries(projectVariants).forEach(
+          ([variantId, textItemsFileContent]) => {
+            const fileName = `${projectId}___${variantId || "base"}`;
+            this.createOutputFile(fileName, variantId, textItemsFileContent);
+          }
+        );
+      }
+    );
+
+    Object.entries(data.componentsMap).forEach(
+      ([variantId, componentsFileContent]) => {
+        const fileName = `components___${variantId || "base"}`;
+        this.createOutputFile(fileName, variantId, componentsFileContent);
+      }
+    );
+
+    return Object.values(this.outputFiles);
+  }
+
+  /**
+   * Sets variants based on configuration
+   * - Fetches from API if "all" configured
+   * - Adds "base" variant by default if none configured
+   */
+  private async fetchVariants(): Promise<void> {
+    let variants: { id: string }[] =
+      this.output.variants ?? this.projectConfig.variants ?? [];
+    if (variants.some((variant) => variant.id === "all")) {
+      variants = await fetchVariants(this.meta);
+    } else if (variants.length === 0) {
+      variants = [{ id: "base" }];
+    }
+
+    this.variants = variants;
+  }
+
+  /**
+   * Fetches text item data via API for each configured project and variant
+   * in this output
+   *
+   * @returns text items mapped to their respective variant and project
+   */
+  private async fetchTextItemsMap(): Promise<TextItemsMap> {
+    if (!this.projectConfig.projects && !this.output.projects) return {};
+    let projects: { id: string }[] =
+      this.output.projects ?? this.projectConfig.projects ?? [];
+
+    const result: TextItemsMap = {};
+
+    if (projects.length === 0) {
+      projects = await fetchProjects(this.meta);
+    }
+
+    for (const project of projects) {
+      result[project.id] = {};
+
+      for (const variant of this.variants) {
+        // map "base" to undefined, as by default export endpoint returns base variant
+        const variantsParam =
+          variant.id === "base" ? undefined : [{ id: variant.id }];
+        const params: PullQueryParams = {
+          ...super.generateQueryParams({
+            projects: [{ id: project.id }],
+            variants: variantsParam,
+          }),
+          format: this.exportFormat,
+        };
+        const textItemsFileContent = await fetchText<TTextItemsResponse>(
```

Comment:
Can we parallelize this one as well by pushing to a promises array and awaiting them all at once?